crates/circuit/src/circuit_data.rs

```diff
@@ -1268,6 +1269,11 @@ impl CircuitData {
         self.qargs_interner().get(index)
     }
 
+    /// Insert qargs into the interner and return the interned value
+    pub fn set_qargs(&mut self, qubits: &[Qubit]) -> Interned<[Qubit]> {
```

Comment:
I thought `set_qargs` was used already in `DAGCircuit`. But I don't see it in there. It made sense to me because this pairs with the `get_qargs` method used for looking up from the interner. But I can change it if you prefer `add_qargs`.

Reply:
Yeah I think `add` or `insert` would be more intuitive 🙂

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/13331/commits/46ac27320c20cbc2ab5461b02e1e1e8c5be07679